core/state_transition.go

```diff
@@ -442,15 +464,28 @@ func (st *stateTransition) execute() (*ExecutionResult, error) {
 	// - reset transient storage(eip 1153)
 	st.state.Prepare(rules, msg.From, st.evm.Context.Coinbase, msg.To, vm.ActivePrecompiles(rules), msg.AccessList)
 
+	if !contractCreation {
+		// Increment the nonce for the next transaction.
+		// Note: EIP-7702 authorizations can also modify the nonce. We perform
+		// this update first to ensure correct validation of authorization nonces.
+		st.state.SetNonce(msg.From, st.state.GetNonce(msg.From)+1)
+	}
+
+	// Apply EIP-7702 authorizations.
+	if msg.AuthList != nil {
+		for _, auth := range msg.AuthList {
+			// Note errors are ignored, we simply skip invalid authorizations here.
+			st.applyAuthorization(msg, &auth)
```

Comment:
EIP says:

```
The transaction is also considered invalid when any field in an authorization tuple cannot fit within the following bounds:

assert auth.chain_id < 2**64
assert auth.nonce < 2**64
assert len(auth.address) == 20
assert auth.y_parity < 2**8
assert auth.r < 2**256
assert auth.s < 2**256
```

so this check should be moved into preCheck

```
	// Limit nonce to 2^64-1 per EIP-2681.
	if auth.Nonce+1 < auth.Nonce {
		return authority, ErrAuthorizationNonceOverflow
	}
```

Reply:
actually the nonce overflow checking is also required in auth execution

EIP: `Verify the nonce is less than 2**64 - 1.`

So the EIP is a bit confusing, either check the nonce overflow before auth execution, or check the nonce overflow in auth execution and simply ignore the invalid auth. @lightclient We should probably improve the EIP

Reply:
The behavior we want is:

- if nonce is >= 2**64, throw away the tx as invalid
- if the nonce is == 2**64-1, the tx is valid but the auth is skipped as invalid
- if the nonce is < 2**64-1, the tx is valid and the auth is valid (assuming all other conditions are met)


Do you have a preferred way to improve the EIP? Generally I tried to separate the tx validity concerns in the bound definitions and the auth validity concerns within the "behavior" seciton.

Reply:
Thanks for the clarification! I was wrong with the conditions (thought the conditions were same). Then it makes sense to me!